streamlit_demo_apps/README.md

```diff
@@ -1,49 +1,77 @@
 # Deploying Streamlit Apps
 
-Deployed Streamlit apps live here in the streamlit_demo_apps folder.
+Follow the steps to deploy a live Streamlit app for your Nada program and test file. The app will connect to the Nillion Testnet to store your Nada program, store secret inputs (or use computation time secrets), and run blind computation.
 
 ## How to add a new Streamlit App
 
-### 0. Create a streamlit secrets file and add your nilchain private key within `.streamlit/secrets.toml`
+### 0. Create a streamlit secrets file
+
+Run this command to create a `.streamlit/secrets.toml` copied from the example.
 
 ```
 cp .streamlit/secrets.toml.example .streamlit/secrets.toml
 ```
 
-### 1. Create an app file in the streamlit_demo_apps folder
-
-Check out the addition app file example:
-
-`app_addition.py`
+Add your Nilchain private key to the .streamlit/secrets.toml file. The private key must be linked to a funded Nillion Testnet address that was created using a Google account (not a mnemonic). This allows you to retrieve the private key from Keplr. If you don’t have a Testnet wallet yet, you can learn how to create one here: https://docs.nillion.com/testnet-guides
 
-### 2. Copy the compiled Nada program files from the target/ folder into the streamlit_demo_apps/compiled_nada_programs folder
+### 1. Run the script to generate a new streamlit app for your program
 
-Check out the compiled Nada program files for addition:
+From the root folder of this repo, run the generate-streamlit-app script:
 
-nada binary `addition.nada.bin`
-nada json `addition.nada.json`
+```
+python3 generate-streamlit-app.py
+```
 
-### 3. Update your app file with the corresponding program name and program test name
+### 2. Follow the prompts to
 
-Check out the addition app file example:
+- Select an existing program (from the src/ directory)
+- Select an existing yaml test file for your program (from the tests/ directory)
 
-`app_addition.py`
+This will generate a Streamlit app file: streamlit*demo_apps/app*[your_program_name].py. The script will run the Streamlit app locally with this command
 
 ```
-program_name = 'addition'
-program_test_name = 'addition_test'
+streamlit run streamlit_demo_apps/app_[your_program_name].py`
 ```
 
-### 4. Test your Streamlit app locally
+### 3. Test your Streamlit app locally
 
-Make sure the apps will work when deployed by testing this command from the root folder.
+View the app in your browser to make sure everything works as expected.
+
+### 4. Commit your code to GitHub
+
+Add and commit your new streamlit app code to your forked Github repo. (Code must be connected to a remote, open source GitHub repository to deploy a Streamlit app.)
```

Comment:
Forking repo was not mentioned in the main repo README / here. Should we mention this higher in case?

With my test - i cloned repo instead of forking it 😅